lib/automations/assign-milestone/README.md

```diff
@@ -0,0 +1,37 @@
+## Assign Milestone Automation
+
+When a pull request is approved and is not already assigned a milestone, this automation will assign the next milestone to it
+automatically.
+
+## Usage
+
+To implement this action, include it in your workflow configuration file:
+
+```yaml
+on:
+    pull_request:
+        types: [opened, synchronize, closed]
+    push:
+    issues:
+        types: [edited]
+jobs:
+    pull-request-automation:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: woocommerce/automations@v1
```

Comment:
I wonder if we should release a `@v2` as a result of implementing the new automation (that way if there are unintended issues with existing automations folks can rollback)?

This can be done in a followup if we end up doing that.

Reply:
Yup I'm fine with that. This shouldn't affect existing workflows because you'd need to call assign-milestone by name in your workflow file?

Reply:
Oh true, I guess there's not any changes to existing tasks (or shared code across tasks) so it's safe to release under the existing version. We could probably reserve version bumps for when there's potentially task breaking changes.

Are you okay with doing a release or want me to take care of it? If you take care of it, this repo uses the release PR automation (similar to process we follow for blocks repo) except the checklist is _much_ smaller. It's currently mostly manual as it's mostly just creating the correct tags etc. The release PR checklist should have all the steps needed.